ibis/selectors.py

```diff
@@ -159,8 +128,17 @@ def numeric() -> Predicate:
     return of_type(dt.Numeric)
 
 
+class OfType(Selector):
+    predicate: Callable[[dt.DataType], bool]
+
+    def expand_names(self, table: ir.Table) -> frozenset[str]:
+        return frozenset(
+            name for name, typ in table.schema().items() if self.predicate(typ)
```

Comment:
This case should be a bit faster too, since we're not going through `Table.__getitem__` anymore, we're just looking at the schema, since we only care about types.